src/components/application_manager/src/state_controller_impl.cc

```diff
@@ -40,15 +42,18 @@ namespace application_manager {
 
 CREATE_LOGGERPTR_GLOBAL(logger_, "StateControllerImpl")
 
-bool IsStatusChanged(HmiStatePtr old_state, HmiStatePtr new_state) {
-  if (old_state->hmi_level() != new_state->hmi_level() ||
-      old_state->audio_streaming_state() !=
-          new_state->audio_streaming_state() ||
-      old_state->system_context() != new_state->system_context()) {
-    return true;
-  }
-  return false;
+namespace {
+bool IsStateChanged(const HmiState& old_state, const HmiState& new_state) {
+  return std::make_tuple(old_state.hmi_level(),
```

Comment:
what's the benefit of using std::make_tuple() !=  std::make_tuple() comparing to the (condition 1 || condition 2 || condition 3 || condition 4) here?
I thought it can only make final binary code bigger.

Reply:
@yang1070 , this is common technik(since C++11) to compare complex objects, to make it easier to write a correct operators(==,!=,<,<=,>,>=) than rolling it yourself, to provide more readable code.
http://en.cppreference.com/w/cpp/utility/tuple/operator_cmp